move/seal-policy/sources/seal_policy.move

```diff
@@ -0,0 +1,27 @@
+// Copyright (c), Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module app::seal_policy;
+
+use enclave::enclave::Enclave;
+use sui::hash::blake2b256;
+
+const ENoAccess: u64 = 0;
+
+entry fun seal_approve<T: drop>(_id: vector<u8>, enclave: &Enclave<T>, ctx: &TxContext) {
```

Comment:
shouldn't this be 
```suggestion
entry fun seal_approve(_id: vector<u8>, enclave: &Enclave<WEATHER>, ctx: &TxContext) {
```

Reply:
hm this is not sufficient that the enclave code (committed with build sha) to the weather type when building ptb?

Reply:
the generic T comes from the enclave type. its added as a OTW so only the contract deployer can register enclave. in the enclave code, the PTB is construct with the type so the enclave code is committed to WEATHER: https://github.com/MystenLabs/nautilus/blob/5a63b86c092df1e255b9593d2d160269cd5df6d0/src/nautilus-server/src/apps/seal-example/endpoints.rs#L204

we could add WEATHER to the move function signature too, to prevent other enclave of other types to call it

Reply:
as the function is currently implement, anyone can create Enclave<DIFFERENT_W> and be able to pass the policy check, as it's not bound to a specific enclave instance or config.

if we want to keep this specific code generic, it should accept also the config for the same T, and use the config's id as the only approved id - this way we restrict it to the right enclaves.
alternatively we can restrict it using the witness maybe (?)